lib/EasyPost/Billing.php

```diff
@@ -53,11 +40,37 @@ public static function fund($amount, $primary_or_secondary = 'primary', $api_key
      * @param string $api_key
      * @return mixed
      */
-    public static function delete($credit_card_id, $api_key = null)
+    public static function delete($primary_or_secondary, $api_key = null)
     {
-        $url = self::classUrl(get_class()) . "/$credit_card_id";
+        $payment_method_info = Billing::get_payment_info($primary_or_secondary);
+
+        $url = $payment_method_info[0] . "/$payment_method_info[1]";
         $requestor = new Requestor($api_key);
         list($response, $api_key) = $requestor->request('delete', $url);
         return Util::convertToEasyPostObject($response, $api_key);
     }
+
+    public static function get_payment_info($primary_or_secondary = 'primary')
```

Comment:
Issue: This is an internal function that our public functions use under the hood and as such, it should not be publicly exposed.

```suggestion
    private static function get_payment_info($primary_or_secondary = 'primary')
```

Reply:
resolved